src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {
+  ChangeEvent, KeyboardEvent, FormEvent, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { CustomForm } from '../CustomForm';
+
+type Props = {
+  todo: Todo;
+  onDelete: (todoId: number) => void,
+  onUpdate: (id: number, data: Partial<Todo>) => void;
+  loadedTodoIds: number[],
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  onUpdate,
+  loadedTodoIds,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [changedTitle, setChangedTitle] = useState(title);
+  const [isEdit, setIsEdit] = useState(false);
+
+  const isActiveTodo = loadedTodoIds.includes(id);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleIsCompletedChange = () => {
+    onUpdate(id, { completed: !completed });
+  };
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setChangedTitle(event.target.value);
+  };
+
+  const handleEditChange = () => {
+    setIsEdit(prevState => !prevState);
+  };
+
+  const closeEditMode = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setChangedTitle(title);
+      setIsEdit(false);
+    }
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!changedTitle.trim()) {
+      onDelete(id);
+    } else if (title !== changedTitle) {
+      onUpdate(id, { title: changedTitle });
+    }
+
+    setIsEdit(false);
+  };
+
+  useEffect(() => {
+    if (inputRef.current !== null) {
+      inputRef.current.focus();
+    }
+  }, [isEdit]);
```

Comment:
```suggestion
  useEffect(() => {
      inputRef.current?.focus();
  }, [isEdit]);
```
can we rewrite it like that?